src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlSerializationILGen.cs

```diff
@@ -421,7 +422,7 @@ private FieldBuilder GenerateTypedSerializers(Hashtable serializers, TypeBuilder
 
             foreach (string key in serializers.Keys)
```

Comment:
Nit: Sorry for not calling attention to this earlier, but this could be improved as follows:

```csharp
foreach (KeyValuePair<string, string> key in serializers)
{
    ...
}
```

 - Avoids the `KeyCollection` allocation for `Keys`
 - Avoids having to lookup the value from the key

There are a few other places throughout that this applies to.

Could be addressed in a subsequent PR, if you prefer.